admin_ui/backend/main.py

```
@@ -5,10 +5,98 @@
+    try:
+        # Prefer explicit env var; fallback matches OutboundStore default.
+        db_path = (os.getenv("CALL_HISTORY_DB_PATH") or "data/call_history.db").strip() or "data/call_history.db"
+        p = Path(db_path)
+        parent = p.parent
+        if parent:
+            try:
+                parent.mkdir(parents=True, exist_ok=True)
+            except Exception:
+                pass
+            try:
+                os.chmod(str(parent), 0o2775)
```

Ruff (0.14.10)

`os.chmod` setting a permissive mask `0o2775` on file or directory

(S103)